app/audit/crypto.ts

```diff
@@ -93,41 +104,44 @@ export const decryptWithPrivateKey = (
  * with the given RSA public key, and returns both values in hex encoding separated by ASYM_SEPARATOR.
  *
  * @param data the plaintext data to encrypt
- * @param publicKey an RSA public key in PEM format
+ * @param scheme object containing the RSA public key in PEM format and a version prefix
  * @return {string} hex-encoded encrypted symmetric key + ASYM_SEPARATOR + hex-encoded encrypted data
  */
-export const encryptData = (data: string, publicKey: Buffer) => {
+export const encryptData = (data: string, scheme: AuditLogScheme) => {
+  const publicKey = Buffer.from(scheme.key);
   const symKey = crypto.randomBytes(CIPHER_BLOCK_SIZE);
   const encryptedData = encryptSym(symKey, data);
   const encryptedSymKey = encryptWithPublicKey(
     publicKey,
     symKey.toString(OUTPUT_ENCODING)
   );
 
-  return [Version.AES_128_GCM_RSA_4096, encryptedSymKey, encryptedData].join(
-    ASYM_SEPARATOR
-  );
+  return [scheme.version, encryptedSymKey, encryptedData].join(ASYM_SEPARATOR);
 };
 
 /**
  * Decrypts the data encrypted with {@link encryptData} using the corresponding private key.
  *
+ * Note: This is not automation ready! Do not use in production.
+ *
  * @param encryptedBlock hex-encoded encrypted symmetric key + ASYM_SEPARATOR + hex-encoded encrypted data
  * @param privateKey the RSA private key for the public key used to encrypt data
  * @return {string} plaintext data
  */
 export const decryptData = (encryptedBlock: string, privateKey: Buffer) => {
-  const [version, encryptedSymKey, encryptedData] =
+  const [, encryptedSymKey, encryptedData] =
```

Comment:
Why did you decide against a differentiation between V1 and V2 here?

Reply:
Up until now, the function was implemented more as a PoC and also to be used in tests and to decrypt data "offline". Also, there was one algorithm and one version to check. But now things are getting more complicated and the function is likely going to be modified in the migration implementation to be (re)used online. Exactly how it's going to be reused and what its logic will look like will depend on the concrete implementation of the migration logic and I felt that any extra logic here might be premature. Moreover, I thought the simple check would give off a false sense of security in this premature state.

Reply:
I did not see this as the PoC but already as a kind of helper function but it makes sense to me.